src/event.rs

```diff
@@ -58,6 +58,23 @@ impl<'a, N: Notify + 'a> input::ActionContext for ActionContext<'a, N> {
 
     fn scroll(&mut self, scroll: Scroll) {
         self.terminal.scroll_display(scroll);
+
+        let (x, y) = (self.mouse().x, self.mouse().y);
+        let size_info = self.size_info();
+        let point = size_info.pixels_to_coords(x, y);
+
+        let modifiers = self.last_modifiers();
+        let motion_mode = TermMode::MOUSE_MOTION | TermMode::MOUSE_DRAG;
+        let report_mode = TermMode::MOUSE_REPORT_CLICK | motion_mode;
+        if self.mouse().left_button_state == ElementState::Pressed &&
+            (modifiers.shift || !self.terminal_mode().intersects(report_mode))
+        {
+            let cell_side = self.get_mouse_side();
+            self.update_selection(Point {
+                line: point.line,
+                col: point.col
+            }, cell_side);
+        }
```

Comment:
Since we should have access to the selection on the terminal from the `scroll` method, it should be possible to skip the check for mouse button states and terminal modes etc.

You could check if a selection is present, but since that is already checked in the `update_selection` method, it would probably be sufficient for functionality if you just call the `update_selection` method without checking for anything.